pyi.py

```diff
@@ -1033,8 +1213,11 @@ def visit_AsyncFunctionDef(self, node: ast.AsyncFunctionDef) -> None:
         if self.in_class.active:
             classdef = self.current_class_node
             assert classdef is not None
+            method_name = node.name
             if _has_bad_hardcoded_returns(node, classdef=classdef):
                 self._Y034_error(node=node, cls_name=classdef.name)
+            elif method_name == "__aexit__":
+                self._check_exit_method(node=node, method_name=method_name)
```

Comment:
I think this doesn't check non-async `__aexit__` methods that return a `Future`.